tool/tctl/common/collection_test.go

```diff
@@ -431,3 +437,44 @@ func makeTestLabels(extraStaticLabels map[string]string) map[string]string {
 	maps.Copy(labels, extraStaticLabels)
 	return labels
 }
+
+// This test makes sure we marshal and unmarshal proto-based Resource153 properly.
+// We had a bug where types.Resource153 implemented by protobuf structs were not
```

Comment:
> We had a bug 

Arguable whether it's a bug, given that Teleport does this at-large and changing the behavior is likely to cause bugs on itself.

Reply:
Before the change, the following command chain failed:
```
tctl get autoupdate_config --format yaml > config.yaml
tctl create -f config.yaml
```

`tctl get`'s output not being accepted by `tctl create` will be considered a bug by the user.

Reply:
It works as intended for various other types, so I think the blanket "it's a bug" is what tripped me. Some qualification is in order, otherwise why wouldn't we just fix the existing "Resource153ToLegacy" method?

Reply:
> why wouldn't we just fix the existing "Resource153ToLegacy" method?

I absolutely want to do this, rending timestamps as objects with seconds/nanos is an inconsistency comparted to what we're doing for every other resource.I'm not doing this because I don't know who else calls `json.Marshal()` and I'm afraid someone might be relying on the wrong behaviour (e.g. we put a json containing the resource in audit logs, the elasticsearch mappings now expect the broken format).